Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php

```diff
@@ -216,18 +217,15 @@ protected function requireNodeTypeToAllowNodesOfTypeInReference(
         if (is_null($propertyDeclaration)) {
             throw ReferenceCannotBeSet::becauseTheNodeTypeDoesNotDeclareIt($referenceName, $nodeTypeName);
         }
-        if (isset($propertyDeclaration['constraints']['nodeTypes'])) {
-            $nodeTypeConstraints = NodeTypeConstraintsWithSubNodeTypes::createFromNodeTypeDeclaration(
-                $propertyDeclaration['constraints']['nodeTypes'],
-                $this->getNodeTypeManager()
+
+        $constraints = $propertyDeclaration['constraints']['nodeTypes'] ?? [];
+
+        if (!ConstraintCheck::create($constraints)->isNodeTypeAllowed($nodeType)) {
```

Comment:
What is the behavior of an empty/missing `constraints` config?

Reply:
```suggestion
        if ($this->constraints === []) {
            return true;
        }
```

yes works .. and this behavior is also covered by childNodes tests